dev_guide/migrating_applications/S2I_tool.adoc

```diff
@@ -0,0 +1,63 @@
+[[dev-guide-s2i-tool]]
+= S2I Tool
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+xref:../../using_images/s2i_images/index.adoc#using-images-s2i-images-index][The
+Source-to-Image (S2I) tool] injects application source code into a Docker image
+and the final product is a new and ready-to-run Docker image that incorporates
+the builder image and built source code. The S2I tool can be installed in your
+local machine without {product-title} from
+link:https://github.com/openshift/source-to-image#installation[the repository].
+
+The S2I tool is a very powerful tool to test and verify your application and
+images locally before using them on {product-title}.
+
+[[s2i-creating-docker-image]]
+== Creating a Docker Image
+
+
+. Identify the builder image that is needed for the application. Red Hat offers
+multiple builder images for different languages including
+link:https://github.com/openshift?utf8=%E2%9C%93&query=s2i[Python, Ruby, Perl,
+PHP, and Node.js]. Other images are available from
+link:https://github.com/openshift-s2i[the community space].
+
+. S2I can build images from source code in a local file system or from a Git repository. To build a
+new Docker image from the builder image and the source code:
++
+----
+$ s2i build <source-location> <builder-image-name> <output-image-name>
+----
++
+[NOTE]
+====
+`<source-location>` can either be a Git repository URL or a directory to source code in a
+local file system.
+====
+
+. Test the built image with Docker daemon:
++
+----
+$ docker run -d --name <new-name> -p <port-number>:<port-number> <output-image-name>
+$ curl localhost:<port-number>
+----
+. Push the new image to the OpenShift registry or Docker hub:
+.. link:https://docs.docker.com/linux/step_six/[Docker registry]
```

Comment:
Now I think about it, I would like to change the text to "Push the new image to OpenShift registry:" (get rid of Docker Hub). Then, remove the Docker reference link. This doc is for OpenShift after all. I think it's best to avoid mentioning Docker Hub. Thanks :).